src/lib.rs

```diff
@@ -172,61 +216,48 @@ pub(crate) const LAKE_FRAMEWORK: &str = "near_lake_framework";
 ///
-/// # asyn fn main() {
-/// let config = LakeConfig {
-///     s3_endpoint: "http://0.0.0.0:9000".to_string(), // AWS S3 custom API endpoint
-///     s3_bucket_name: "near-lake-custom".to_string(), // AWS S3 bucket name
-///     s3_region_name: "eu-central-1".to_string(), // AWS S3 bucket region
-///     start_block_height: 1, // the latest block height
-/// };
-/// let stream = near_lake_framework::streamer(config);
+/// # async fn main() {
+///    let config = LakeConfigBuilder::default()
+///        .testnet()
+///        .start_block_height(82422587)
+///        .build()
+///        .expect("Failed to build LakeConfig");
+///
+///     let stream = near_lake_framework::streamer(config);
 ///
-/// while let Some(streamer_message) = stream.recv().await {
-///     eprintln!("{:#?}", streamer_message);
-/// }
+///     while let Some(streamer_message) = stream.recv().await {
+///         eprintln!("{:#?}", streamer_message);
+///     }
 /// # }
 /// ```
 pub fn streamer(config: LakeConfig) -> mpsc::Receiver<near_indexer_primitives::StreamerMessage> {
     let (sender, receiver) = mpsc::channel(100);
-    tokio::spawn(start(
-        sender,
-        config.s3_endpoint,
-        config.s3_bucket_name,
-        config.s3_region_name,
-        config.start_block_height,
-    ));
+    tokio::spawn(start(sender, config));
     receiver
 }
 
 async fn start(
     streamer_message_sink: mpsc::Sender<near_indexer_primitives::StreamerMessage>,
-    s3_endpoint: Option<String>,
-    s3_bucket_name: String,
-    s3_region_name: String,
-    index_from_block_height: types::BlockHeight,
+    config: LakeConfig,
 ) {
-    // instantiate AWS S3 Client
-    let region_provider = RegionProviderChain::first_try(Some(s3_region_name).map(Region::new))
-        .or_default_provider()
-        .or_else(Region::new("eu-central-1"));
-    let shared_config = aws_config::from_env().region(region_provider).load().await;
-    let mut s3_conf = aws_sdk_s3::config::Builder::from(&shared_config);
-    // Owerride S3 endpoint in case you want to use custom solution
-    // like Minio or Localstack as a S3 compatible storage
-    if let Some(endpoint) = s3_endpoint {
-        s3_conf = s3_conf.endpoint_resolver(Endpoint::immutable(endpoint.parse::<Uri>().unwrap()));
-        tracing::info!(
-            target: LAKE_FRAMEWORK,
-            "Custom S3 endpoint used: {}",
-            endpoint
-        );
-    }
-    let s3_client = Client::from_conf(s3_conf.build());
+    let mut start_from_block_height = config.start_block_height;
+    let s3_bucket_name = config
+        .s3_bucket_name
+        .expect("`s3_bucket_name` expected to be set by this time");
+
+    let s3_client = if let Some(config) = config.s3_config {
+        Client::from_conf(config)
+    } else {
+        let shared_config = aws_config::from_env().load().await;
+        let s3_config = aws_sdk_s3::config::Builder::from(&shared_config)
+            .region(aws_types::region::Region::new("eu-central-1"))
```

Comment:
Can we avoid this hard-coding? I feel it is out of place here

Reply:
At the moment it is the only supported region. I can introduce enum with available regions (thought with only one variant for now) and replace this line with something like:

```rust
.region(config.region)
```

What do you think? Alternatively, we can:
1. keep it as is until we provide more region support with the list of available
2. return the `s3_region_name` parameter to `LakeConfig` and pass it here

Reply:
> return the s3_region_name parameter to LakeConfig and pass it here

I think `testnet` and `mainnet` could set the relevant value. Having `s3_region_name` configurable, makes it flexible.